imod_coupler/drivers/metamod/metamod.py

```diff
@@ -63,109 +60,47 @@ class MetaMod(Driver):
+    def __init__(self, base_config: BaseConfig, metamod_config: MetaModConfig):
         """Constructs the `MetaMod` object"""
-        self.config = config
-        self.config_path = config_path
+        self.base_config = base_config
+        self.metamod_config = metamod_config
+        self.coupling = metamod_config.coupling[
+            0
+        ]  # Adapt as soon as we have multimodel support
 
     def initialize(self) -> None:
-        self.set_vars_from_config()
-
+        self.mf6 = XmiWrapper(
+            lib_path=self.metamod_config.kernels.modflow6.dll,
+            lib_dependency=self.metamod_config.kernels.modflow6.dll_dep_dir,
+            working_directory=self.metamod_config.kernels.modflow6.work_dir,
+            timing=self.base_config.timing,
+        )
+        self.msw = XmiWrapper(
+            lib_path=self.metamod_config.kernels.metaswap.dll,
+            lib_dependency=self.metamod_config.kernels.metaswap.dll_dep_dir,
+            working_directory=self.metamod_config.kernels.metaswap.work_dir,
+            timing=self.base_config.timing,
+        )
         # Print output to stdout
         self.mf6.set_int("ISTDOUTTOFILE", 0)
         self.mf6.initialize()
         self.msw.initialize()
         self.couple()
 
-    def set_vars_from_config(self) -> None:
-        self.mf6 = self.initialize_kernel_from_config("modflow6")
-        self.msw = self.initialize_kernel_from_config("metaswap")
-
-        # TODO: Relax this requirement as soon as we support multiple models
-        assert len(self.config["driver"]["coupling"]) == 1
-        coupling_config = self.config["driver"]["coupling"][0]
-
-        self.mf6_model = coupling_config["mf6_model"]
-        self.mf6_msw_recharge_pkg = coupling_config["mf6_msw_recharge_pkg"]
-        self.mf6_msw_node_map = self.get_absolute_path(
-            coupling_config["mf6_msw_node_map"]
-        )
-        if not self.mf6_msw_node_map.is_file():
-            raise ValueError(
-                f"'{self.mf6_msw_node_map=}' is not a valid path to a file."
-            )
-        self.mf6_msw_recharge_map = self.get_absolute_path(
-            coupling_config["mf6_msw_recharge_map"]
-        )
-        if not self.mf6_msw_recharge_map.is_file():
-            raise ValueError(
-                f"'{self.mf6_msw_recharge_map=}' is not a valid path to a file."
-            )
-
-        self.is_sprinkling_active = coupling_config["enable_sprinkling"]
-        if self.is_sprinkling_active:
-            self.mf6_msw_sprinkling_map = self.get_absolute_path(
-                coupling_config["mf6_msw_sprinkling_map"]
-            )
-            if not self.mf6_msw_sprinkling_map.is_file():
-                raise ValueError(
-                    f"'{self.mf6_msw_sprinkling_map=}' is not a valid path to a file."
-                )
-            self.mf6_msw_well_pkg = coupling_config["mf6_msw_well_pkg"]
-
-    def get_absolute_path(self, path: str | Path) -> Path:
```

Comment:
Are we sure that absolute paths are not necessary anymore after the os.chdir?
It now seems we throw away some logic that had a use before and might still have.